src/pages/details/gcpDetails/gcpDetails.tsx

```diff
@@ -0,0 +1,519 @@
+
+    if (action === 'none') {
+      this.setState({ isAllSelected: false, selectedItems: [] });
+    } else if (action === 'page') {
+      this.setState({
+        isAllSelected: false,
+        selectedItems: this.getComputedItems(),
+      });
+    } else if (action === 'all') {
+      this.setState({ isAllSelected: !isAllSelected, selectedItems: [] });
+    }
+  };
+
+  public handleExportModalClose = (isOpen: boolean) => {
+    this.setState({ isExportModalOpen: isOpen });
+  };
+
+  public handleExportModalOpen = () => {
+    this.setState({ isExportModalOpen: true });
+  };
+
+  private handleFilterAdded = (filterType: string, filterValue: string) => {
+    const { history, query } = this.props;
+    const newQuery = { ...JSON.parse(JSON.stringify(query)) };
+
+    // Filter by * won't generate a new request if group_by * already exists
+    if (filterValue === '*' && newQuery.group_by[filterType] === '*') {
+      return;
+    }
+
+    if (newQuery.filter_by[filterType]) {
+      let found = false;
+      const filters = newQuery.filter_by[filterType];
+      if (!Array.isArray(filters)) {
+        found = filterValue === newQuery.filter_by[filterType];
+      } else {
+        for (const filter of filters) {
+          if (filter === filterValue) {
+            found = true;
+            break;
+          }
+        }
+      }
+      if (!found) {
+        newQuery.filter_by[filterType] = [newQuery.filter_by[filterType], filterValue];
+      }
+    } else {
+      newQuery.filter_by[filterType] = [filterValue];
+    }
+    const filteredQuery = this.getRouteForQuery(newQuery, true);
+    history.replace(filteredQuery);
+  };
+
+  private handleFilterRemoved = (filterType: string, filterValue: string) => {
+    const { history, query } = this.props;
+    const newQuery = { ...JSON.parse(JSON.stringify(query)) };
+
+    if (filterType === null) {
+      newQuery.filter_by = undefined; // Clear all
+    } else if (filterValue === null) {
+      newQuery.filter_by[filterType] = undefined; // Clear all values
+    } else if (Array.isArray(newQuery.filter_by[filterType])) {
+      const index = newQuery.filter_by[filterType].indexOf(filterValue);
+      if (index > -1) {
+        newQuery.filter_by[filterType] = [
+          ...query.filter_by[filterType].slice(0, index),
+          ...query.filter_by[filterType].slice(index + 1),
+        ];
+      }
+    } else {
+      newQuery.filter_by[filterType] = undefined;
+    }
+    const filteredQuery = this.getRouteForQuery(newQuery, true);
+    history.replace(filteredQuery);
+  };
+
+  private handleGroupByClick = groupBy => {
+    const { history, query } = this.props;
+
+    let groupByKey = groupBy;
+    let value = '*';
+
+    // Check for for org units
+    const index = groupBy.indexOf(orgUnitIdKey);
+    if (index !== -1) {
+      groupByKey = orgUnitIdKey.substring(0, orgUnitIdKey.length);
+      value = groupBy.slice(orgUnitIdKey.length);
+    }
```

Comment:
Please remove the org unit code. This should not be a group_by or filter option in the details page

Reply:
Should I remove the entire "handleGroupByClick" function here?